### ISSUE TITLE
[BUG] Need Updated CSI-Proxy to Address Windows Volume Mount Issue

### ISSUE THREAD

**Issue by scottstout** [rancher/rancher#42298]
**Rancher Server Setup**
- Rancher version: 2.7.x (whatever is currently in main branch)
- Installation option (Docker install/Helm Chart):
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):RKE2

**Describe the bug**
Search for search for CATTLE_CSI_PROXY_AGENT_VERSION in this: https://github.com/rancher/rancher/blob/release/v2.7/package/Dockerfile
You will see that v1.1.1 is being used. 

There is an issue that is fixed in v1.1.2. The PR for the issue is here: https://github.com/kubernetes-csi/csi-proxy/pull/210

You can see it in the release notes here: https://github.com/kubernetes-csi/csi-proxy/blob/master/CHANGELOG/CHANGELOG-1.1.2.md

Described as "Try to fix issue with 'Multiple connections to a server' https://github.com/kubernetes-csi/csi-proxy/pull/210 ([vitaliy-leschenko](https://github.com/vitaliy-leschenko))"

**Comment by samjustus** [comment]
sure-7848

**Comment by markusewalker** [comment]
Validated that this is addressed on `v2.9-head` https://github.com/rancher/rancher/commit/12a48d56eb2e577ac386134aca1c90aadacd5069

**ENVIRONMENT DETAILS**
- Rancher install: Docker
- Rancher version: `v2.9-head` https://github.com/rancher/rancher/commit/12a48d56eb2e577ac386134aca1c90aadacd5069

**TEST RESULT**
|#|Scenario|Test Case|Result
|---|---|---|---|
|1|CSI proxy v1.1.3 establishes multiple connections to smb share|Create a Windows RKE2 cluster, install csi driver smb chart, create a SMB server and validate that it is working|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. As a standard user, provisioned a standard RKE2 cluster ( 1 etcd, 1 cp, 1 Linux worker, 2 Windows workers).
2. Created new Helm repo using `https://raw.githubusercontent.com/kubernetes-csi/csi-driver-smb/master/charts`.
3. Installed `csi-driver-smb` chart.
4. Followed the following [guide](https://www.phillipsj.net/posts/how-to-use-the-windows-csi-proxy-and-csi-smb-driver-for-kubernetes/) on creating an SMB server for testing.
5. Exec'ed into one of the deployments and validated `data.txt` is available and is not empty:
```
C:\mnt\smb>dir
 Volume in drive C has no label.
 Volume Serial Number is 0813-38F2

 Directory of C:\mnt\smb

03/26/2024  03:42 PM    <DIR>          .
03/26/2024  03:42 PM    <DIR>          ..
03/26/2024  03:43 PM             1,144 data.txt
               1 File(s)          1,144 bytes
               2 Dir(s)  94,809,133,056 bytes free
```
